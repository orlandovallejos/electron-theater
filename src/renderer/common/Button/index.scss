.primary {
  background: linear-gradient(244deg, #068eff 30.91%, #95d8ff 150.35% );
  border: none;
  padding: 14px 50px;
  border-radius: 5px;
  color: #fff;
  font-weight: 400;
  font-size: 16px;
  cursor: pointer;
}

.secondaryWrapper {
  backdrop-filter: blur(5px) contrast(.8);
  border-radius: 5px;

  button.secondary {
    background: none;
  }
}

.secondary {
  composes: primary;
}
