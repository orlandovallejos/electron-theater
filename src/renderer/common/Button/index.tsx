/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
import {
  ButtonPrimary,
  WrapperSecondary,
  ButtonSecondary,
} from 'renderer/common/Button/style';

type Props = {
  type: 'primary' | 'secondary';
  children: React.ReactNode;
};

const Button = (props: Props) => {
  const { type, children } = props;
  let component;

  switch (type) {
    case 'secondary':
      component = (
        <WrapperSecondary>
          <ButtonSecondary>{children}</ButtonSecondary>
        </WrapperSecondary>
      );
      break;
    default:
      component = <ButtonPrimary>{children}</ButtonPrimary>;
      break;
  }

  return component;
};

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export default Button;
