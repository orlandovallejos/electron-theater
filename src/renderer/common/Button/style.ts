/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import styled from 'styled-components';

const getButtonProps = () => `
  border: none;
  padding: 14px 50px;
  border-radius: 5px;
  color: #fff;
  font-weight: 400;
  font-size: 16px;
  cursor: pointer;
  outline: none;
`;

const ButtonPrimary = styled.button`
  ${() => getButtonProps()};
  background: linear-gradient(244deg, #068eff 30.91%, #95d8ff 150.35%);
`;
ButtonPrimary.displayName = 'ButtonPrimary';

const WrapperSecondary = styled.div`
  backdrop-filter: blur(5px) contrast(0.8);
  border-radius: 5px;
`;
const ButtonSecondary = styled.button`
  ${() => getButtonProps()};
  background: none;
`;
ButtonSecondary.displayName = 'ButtonSecondary';

/*------------------------------------------------*/
/* EXPORTS
/*------------------------------------------------*/
export { ButtonPrimary, WrapperSecondary, ButtonSecondary };
export default null;
