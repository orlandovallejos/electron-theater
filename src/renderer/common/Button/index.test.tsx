/*------------------------------------------------*/
/* LIBRARIES
/*------------------------------------------------*/
import React from 'react';
import shallow from 'enzyme/shallow';
/*------------------------------------------------*/
/* INTERNAL DEPENDENCIES
/*------------------------------------------------*/
import Button from './index';

describe('<Button />', () => {
  it('should render a primary button', () => {
    const wrapper = shallow(<Button type="primary">Click me!</Button>);

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
