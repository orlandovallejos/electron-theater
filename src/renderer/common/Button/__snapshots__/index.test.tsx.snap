// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a primary button 1`] = `
"<ButtonPrimary>
  Click me!
</ButtonPrimary>"
`;
